### PR TITLE
PAAS-186 add seed button to deploy group matrix for easy seedin with default v…

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/paas 

with missing role:
screen shot 2016-05-31 at 1 22 29 pm

after clicking:
screen shot 2016-05-31 at 4 13 09 pm

### CODE REVIEW

File: plugins/kubernetes/app/controllers/admin/kubernetes/deploy_group_roles_controller.rb [hunk 1]
```diff
@@ -1,7 +1,10 @@
 class Admin::Kubernetes::DeployGroupRolesController < ApplicationController
-  before_action :authorize_deployer!
+  DEPLOYER_ACCESS = [:index, :show, :new]
```
Comment (grosser) [inline plugins/kubernetes/app/controllers/admin/kubernetes/deploy_group_roles_controller.rb]: switching to whitelist for better security

**Comment by grosser** [comment]
screen shot 2016-05-31 at 4 52 46 pm
screen shot 2016-05-31 at 4 53 21 pm

File: plugins/kubernetes/app/models/kubernetes/deploy_group_role.rb [hunk 2]
```diff
@@ -5,5 +5,44 @@ class DeployGroupRole < ActiveRecord::Base
...
+        dg_roles = project_dg_roles.select { |r| r.deploy_group_id == deploy_group.id }
+        role_pairs = roles.map do |role|
+          [role, dg_roles.detect { |r| r.kubernetes_role_id == role.id }]
+        end
+        [deploy_group, role_pairs]
+      end
+    end
+
+    # add deploy group roles for everything missing from the matrix
+    def self.seed!(stage)
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/deploy_group_role.rb]: Document possible return types

File: plugins/kubernetes/app/models/kubernetes/role.rb [hunk 3]
```diff
@@ -60,6 +60,20 @@ def self.configured_for_project(project, git_sha)
       end
     end
 
+    def defaults
+      return unless raw_template = project.repository.file_content(config_file, 'HEAD', update: false)
+      objects = Array.wrap(Kubernetes::Util.parse_file(raw_template, config_file))
+      return unless deploy = objects.detect { |o| ['Deployment', 'DaemonSet'].include?(o.fetch('kind')) }
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/role.rb]: I think this logic already exists in ReleaseDoc or DeployYAML or something.  DRY if possible?

File: plugins/kubernetes/app/models/kubernetes/role.rb [hunk 4]
```diff
@@ -60,6 +60,20 @@ def self.configured_for_project(project, git_sha)
...
+
+      replicas = deploy['spec']['replicas']
+
+      return unless limits = deploy['spec']['template']['spec']['containers'].first['resources'].try(:[], 'limits')
+      cpu = limits['cpu'].to_i / 1000.0 # 250m -> 0.25
+      ram = limits['ram'].to_i # 200Mi -> 200
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/role.rb]: Not accurate.  Look at https://github.com/kubernetes/kubernetes/blob/master/docs/design/resources.md#resource-types

**Comment by grosser** [comment]
all fixed now

File: plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
 module Kubernetes
   class DeployYaml
     CUSTOM_UNIQUE_LABEL_KEY = 'rc_unique_identifier'.freeze
-    DAEMON_SET = 'DaemonSet'.freeze
-    DEPLOYMENT = 'Deployment'.freeze
+    SIDECAR_NAME = 'secret-sidecar'
+    SIDECAR_IMAGE = ENV['SECRET_SIDECAR_IMAGE']
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb]: How about we add `presence` here, in case someone sets the ENV var to be an empty string?

**Comment by jonmoter** [comment]
👍 on 💚

File: plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb (hunk 5 above)
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb]: added

File: plugins/kubernetes/app/models/kubernetes/deploy_group_role.rb (hunk 2 above)
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/deploy_group_role.rb]: done

**Pull request merged**